src/plugins/data/public/search/fetch/handle_warnings.tsx

```diff
@@ -21,33 +23,56 @@ import {
 } from '../types';
 import { extractWarnings } from './extract_warnings';
 
+const getDebouncedWarning = () => {
+  const addWarning = () => debounce(getNotifications().toasts.addWarning, 10000, { leading: true });
+  const memory: Record<string, ReturnType<typeof addWarning>> = {};
+
+  return (
+    debounceKey: string,
```

Comment:
I think we can use `memoize` and avoid having our own store here

Reply:
we discussed this and decided to leave the code as is.